.github/workflows/rust.yml

```diff
@@ -242,17 +353,76 @@ jobs:
           -p linera-client \
           -p linera-rpc \
           -p linera-views
+
+  cargo-doc:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Clear up some space
+      run: |
+        sudo rm -rf /usr/share/dotnet
+        sudo rm -rf /opt/ghc
+        sudo rm -rf "/usr/local/share/boost"
+        sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+    - name: Put lint toolchain file in place
+      run: |
+        ln -sf toolchains/nightly/rust-toolchain.toml
+    - uses: actions-rust-lang/setup-rust-toolchain@v1
+    - name: Install Protoc
+      uses: arduino/setup-protoc@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
     - name: Run cargo doc
       run: |
         cargo doc --locked --all-features
-    - name: Run cargo check-all-features
-      run: |
-        cargo check-all-features
-        cargo check-all-features --all-targets
-    - name: Restore `rust-toolchain.toml` file
-      if: '!cancelled()'
+
+  check-linera-service-graphql-schema:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Clear up some space
       run: |
-        ln -sf toolchains/stable/rust-toolchain.toml
+        sudo rm -rf /usr/share/dotnet
+        sudo rm -rf /opt/ghc
+        sudo rm -rf "/usr/local/share/boost"
+        sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+    - uses: actions-rust-lang/setup-rust-toolchain@v1
+    - name: Install Protoc
+      uses: arduino/setup-protoc@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
     - name: Check linera-service GraphQL schema
       run: |
         diff <(cargo run --locked --bin linera-schema-export) linera-service-graphql-client/gql/service_schema.graphql
+
+  check-all-features:
+    runs-on: ubuntu-latest-16-cores
+    timeout-minutes: 40
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Clear up some space
+      run: |
+        sudo rm -rf /usr/share/dotnet
+        sudo rm -rf /opt/ghc
+        sudo rm -rf "/usr/local/share/boost"
+        sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+    - name: Put lint toolchain file in place
+      run: |
+        ln -sf toolchains/nightly/rust-toolchain.toml
+    - uses: actions-rust-lang/setup-rust-toolchain@v1
+    - name: Install cargo-all-features
+      run: |
+        cargo install --git https://github.com/ma2bd/cargo-all-features --branch workspace_metadata --locked
+    - name: Install Protoc
+      uses: arduino/setup-protoc@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+    - name: Run cargo check-all-features
+      run: |
+        cargo check-all-features
+        cargo check-all-features --all-targets
```

Comment:
Also, isn't it redundant to run both of these? I'm removing `cargo check-all-features`

Reply:
I don't remember it but I must have had a reason